src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -32,12 +38,14 @@ public void onCompletion(RecordMetadata md, Exception e) {
 			try {
 				if ( LOGGER.isDebugEnabled()) {
 					LOGGER.debug("->  topic:" + md.topic() + ", partition:" +md.partition() + ", offset:" + md.offset());
-					LOGGER.debug("   " + event.toJSON());
+					LOGGER.debug("   " + this.json);
 					LOGGER.debug("   " + event.getNextBinlogPosition());
 					LOGGER.debug("");
 				}
-				config.setInitialPosition(event.getNextBinlogPosition());
-			} catch (IOException | SQLException e1) {
+				if ( this.lastRowInEvent ) {
+					config.setInitialPosition(event.getNextBinlogPosition());
+				}
```

Comment:
yeah, if it fails mid-way through we can repeat rows.  We should do a whole bit on the consistancy guarantees, ie what happens if you change masters, the one second flush window, this, etc...